sdk/src/main/java/io/dapr/client/domain/BulkMessage.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.client.domain;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a bulk of messages received from the message bus.
+ */
+public class BulkMessage<T> {
+  private List<BulkMessageEntry<T>> entries;
+  private String topic;
+  private Map<String, String> metadata;
+
+  /**
+   * Instantiate a BulkMessage.
+   */
+  public BulkMessage() {
```

Comment:
Can you really instantiate  BulkMessage with nothing?

Reply:
I am taking this implementation from #789 and making it immutable.